app/jobs/teacher_training_api/create_courses_job.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module TeacherTrainingApi
+  class CreateCoursesJob < ApplicationJob
+    queue_as :default
+
+    def perform
+      return unless FeatureService.enabled?("create_courses_from_ttapi")
+
+      Provider.all.each do |provider|
+        CreateProviderCoursesJob.perform_later(provider.id)
```

Comment:
This used to be an issue in the past but you can just pass in the provider now and Rails will use `global id` to reference the model instance. https://github.com/rails/globalid. Then you can remove the extra `.find` in the `create_provider_courses_job.rb`.

Reply:
Woah. I did not know this - thanks!